plugin/al/plugin/AL_USDMayaTestPlugin/AL/usdmaya/commands/test_ProxyShapeSelect.cpp

```diff
@@ -57,28 +104,52 @@ TEST(ProxyShapeSelect, selectNode1)
 
   auto compareNodes = [] (const SdfPathVector& paths)
   {
-    MSelectionList sl;
-    MGlobal::getActiveSelectionList(sl);
-    EXPECT_EQ(sl.length(), paths.size());
-    for(uint32_t i = 0; i < sl.length(); ++i)
+    if(!MayaUsdProxyShapePlugin::useVP2_NativeUSD_Rendering())
     {
-      MObject obj;
-      sl.getDependNode(i, obj);
-      MFnDependencyNode fn(obj);
-      MStatus status;
-      MPlug plug = fn.findPlug("primPath", &status);
-      EXPECT_EQ(MStatus(MS::kSuccess), status);
-      MString pathName = plug.asString();
-      bool found = false;
-      for(uint32_t j = 0; j < paths.size(); ++j)
+      MSelectionList sl;
+      MGlobal::getActiveSelectionList(sl);
+      EXPECT_EQ(sl.length(), paths.size());
+
+      for(uint32_t i = 0; i < sl.length(); ++i)
       {
-        if(pathName == paths[j].GetText())
+        MObject obj;
+        sl.getDependNode(i, obj);
+        MFnDependencyNode fn(obj);
+        MStatus status;
+        MPlug plug = fn.findPlug("primPath", &status);
+        EXPECT_EQ(MStatus(MS::kSuccess), status);
+        MString pathName = plug.asString();
+        bool found = false;
+        for(uint32_t j = 0; j < paths.size(); ++j)
         {
-          found = true;
-          break;
+          if(pathName == paths[j].GetText())
+          {
+            found = true;
+            break;
+          }
         }
+        EXPECT_TRUE(found);
+      }
+    }
+    else
+    {
+      auto selection = Ufe::GlobalSelection::get();
+      for(auto it : *selection)
+      {
+        bool found = false;
+        for(uint32_t j = 0; j < paths.size(); ++j)
+        {
+          auto str = it->path().string();
+          auto index = str.find_first_of('/');
```

Comment:
Looks like you're trying to get the second segment in the UFE path, i.e.
it->path().getSegments()[1]